### PR TITLE
Drop most use of private GNOME APIs

### PR DESCRIPTION
**Pull request by jadahl**
This removes the screen-cast, remote-desktop, screenshot and background portal backends, as they all used private APIs that are have no API stability across major GNOME versions.

----

There is still use of `org.gnome.Screensaver` and `org.gnome.SessionManager`, not sure if/when it makes sense to ditch those too, but lets start with those that recently has seen API changes/additions.

### CODE REVIEW

File: data/gtk.portal.in [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [portal]
 DBusName=org.freedesktop.impl.portal.desktop.gtk
-Interfaces=org.freedesktop.impl.portal.FileChooser;@appchooser_iface@org.freedesktop.impl.portal.Print;@screenshot_iface@org.freedesktop.impl.portal.Notification;org.freedesktop.impl.portal.Inhibit;org.freedesktop.impl.portal.Access;org.freedesktop.impl.portal.Account;org.freedesktop.impl.portal.Email;org.freedesktop.impl.portal.DynamicLauncher;@screencast_iface@@remotedesktop_iface@@lockdown_iface@@background_iface@@settings_iface@@wallpaper_iface@
+Interfaces=org.freedesktop.impl.portal.FileChooser;@appchooser_iface@org.freedesktop.impl.portal.Print;@screenshot_iface@org.freedesktop.impl.portal.Notification;org.freedesktop.impl.portal.Inhibit;org.freedesktop.impl.portal.Access;org.freedesktop.impl.portal.Account;org.freedesktop.impl.portal.Email;org.freedesktop.impl.portal.DynamicLauncher;@lockdown_iface@@settings_iface@@wallpaper_iface@
```
Comment (smcv) [inline data/gtk.portal.in:3]: Surely we also want to lose `@screenshot_iface@`?

```suggestion
Interfaces=org.freedesktop.impl.portal.FileChooser;@appchooser_iface@org.freedesktop.impl.portal.Print;org.freedesktop.impl.portal.Notification;org.freedesktop.impl.portal.Inhibit;org.freedesktop.impl.portal.Access;org.freedesktop.impl.portal.Account;org.freedesktop.impl.portal.Email;org.freedesktop.impl.portal.DynamicLauncher;@lockdown_iface@@settings_iface@@wallpaper_iface@
```

**Review by smcv — CHANGES_REQUESTED** [review]

**Comment by smcv** [comment]
This makes sense to me. GNOME users should be using xdp-gnome, or if their GNOME version is too old for that, they can stay on xdp-gtk version 1.14.x.

Debian's xdp-gtk is currently configured to work in parallel with xdp-gnome, and also be suitable for use as a fallback by other desktops. Looking at the optional interfaces, we're currently enabling the appchooser (for OpenURI on otherwise unsupported desktops like XFCE), lockdown (because gvfs uses it), and settings (fonts, antialiasing) interfaces, but we are not enabling the wallpaper interface, in addition to the ones that this PR deletes.

xdp-xapp doesn't seem to have appchooser either, so users of XFCE, Cinnamon, MATE are presumably going to be relying on this backend for it.

File: data/gtk.portal.in (hunk 1 above)
Comment (jadahl) [reply data/gtk.portal.in:3]: Ah, thanks, thought I git grep:ed for screenshot.

File: README.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
 # xdg-desktop-portal-gtk
 
 A backend implementation for [xdg-desktop-portal](http://github.com/flatpak/xdg-desktop-portal)
-that is using GTK and various pieces of GNOME infrastructure, such as the
-org.gnome.Shell.Screenshot or org.gnome.SessionManager D-Bus interfaces.
+that is using GTK and various pieces of GNOME infrastructure, such as
+org.gnome.desktp.* GSettings schemas and the org.gnome.SessionManager and
```
Comment (smcv) [inline README.md:5]:
```suggestion
`org.gnome.desktop.*` GSettings schemas and the org.gnome.SessionManager and
```

**Comment by smcv** [comment]
> Perhaps wallpaper is suitable for torching too? Makes sense to leave the other ones you mention be.

Up to you, I don't really remember what the wallpaper interface interacts with.

If there's any doubt about it, I think it would make more sense to get rid of these ones and keep wallpaper than waiting for a decision on wallpaper's fate - we can always remove that one as well, later.

**Comment by jadahl** [comment]
The wallpaper just sets a `org.gnome.desktop.background` entry to something, but I'd leave it there for now, to not stall anything else.

**Review by smcv — APPROVED** [review]

**Pull request merged**